### PR TITLE
Add create rollup job api to high level rest client

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds the Create Rollup Job API to the high level REST client. It supersedes #32703 and adds dedicated request/response objects so that it does not depend on server side components.

Related #29827

### CODE REVIEW

**Comment by tlrx** [comment]
@polyfractal I'm sorry to create a new pull request for this but that was really easier as I created new classes for request, response and configurations objects. I kept the parsing logic even though it's not strictly required for this API as it simplifies the unit tests and prepare the field for other Rollup APIs.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.apache.http.client.methods.HttpPut;
+import org.elasticsearch.client.rollup.PutRollupJobRequest;
+
+import java.io.IOException;
+
+import static org.elasticsearch.client.RequestConverters.REQUEST_BODY_CONTENT_TYPE;
+import static org.elasticsearch.client.RequestConverters.createEntity;
+
+final class RollupRequestConverters {
```
Comment (polyfractal) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: ++ to a dedicated class for these Rollup converters

**Review by polyfractal — APPROVED** [review]
Thanks @tlrx!  Sorry for the delay on getting this reviewed, busy week :)

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ValidationException.java [hunk 2]
```diff
@@ -31,10 +33,23 @@
...
+    public void addValidationError(final String error) {
         validationErrors.add(error);
     }
 
+    /**
+     * Adds validation errors from an existing {@link ValidationException} to
+     * the accumulating validation errors
+     * @param exception the {@link ValidationException} to add errors from
+     */
+    public final void addValidationErrors(final @Nullable ValidationException exception) {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ValidationException.java]: excellent

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/PutRollupJobResponse.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+public class PutRollupJobResponse implements ToXContentObject {
+
+    private final boolean acknowledged;
+
+    public PutRollupJobResponse(final boolean acknowledged) {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/PutRollupJobResponse.java]: Im ok with making a common response in client for the Ack Responses. We have a lot of them.

**Review by hub-cap — APPROVED** [review]
LGTM, but its huge. I cant really do any proper reviewing on the sheer number of classes moving over, so I have to trust that its sane. Even breaking it up will only get u about 2k for the POJOs and like 1k for the HLRC stuff. Still huge.

Great work on the validatable stuff tho.

**Comment by tlrx** [comment]
Thanks @polyfractal and @hub-cap!

I apologize for the size of the pull request; I tried to split it into multiple chunks (like I did to clean up the config objects) but did not succeed.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/PutRollupJobResponse.java (hunk 3 above)
Comment (tlrx) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/PutRollupJobResponse.java]: I agree, but the PR was already large enough. I think it must be done once another ack response is added to the HLRC.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RollupIT.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        RefreshResponse refreshResponse = highLevelClient().indices().refresh(new RefreshRequest("docs"), RequestOptions.DEFAULT);
+        assertEquals(0, refreshResponse.getFailedShards());
+
+        final String id = randomAlphaOfLength(10);
+        final String indexPattern = randomFrom("docs", "d*", "doc*");
+        final String rollupIndex = randomFrom("rollup", "test");
+        final String cron = "*/1 * * * * ?";
+        final int pageSize = randomIntBetween(numDocs, numDocs * 10);
+        // TODO expand this to also test with histogram and terms?
+        final GroupConfig groups = new GroupConfig(new DateHistogramGroupConfig("date", DateHistogramInterval.DAY));
```
[the remark this answers is not in the document]
Comment (polyfractal) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/RollupIT.java]: :+1:

[the remark this answers is not in the document]
Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/RollupIT.java]: <3

[the remark this answers is not in the document]
Comment (tlrx) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/RollupIT.java]: Thanks @polyfractal

**Pull request merged**